pkg/package-server-manager/controller.go

```diff
@@ -98,22 +99,35 @@ func reconcileCSV(log logr.Logger, image string, csv *olmv1alpha1.ClusterService
 	if csv.ObjectMeta.CreationTimestamp.IsZero() {
 		log.Info("attempting to create the packageserver csv")
 	}
-	if !validateCSV(log, csv) {
-		log.Info("updating invalid csv to use the default configuration")
-		tmp, err := manifests.NewPackageServerCSV(
-			manifests.WithName(csv.Name),
-			manifests.WithNamespace(csv.Namespace),
-			manifests.WithImage(image),
-		)
-		if err != nil {
-			return err
-		}
-		csv.Spec = tmp.Spec
+
+	// The Package Server CSV is defined within this repo as a yaml manifest which
+	// acts as a "base" of the CSV we expect. We then apply the highAvailability
+	// configurations with the `ensureCSV` method to the expectedCSV object. The
+	// csv passed into this function then has the expected labels, annotations,
+	// and spec set.
+	expectedCSV, err := manifests.NewPackageServerCSV(
+		manifests.WithName(csv.Name),
+		manifests.WithNamespace(csv.Namespace),
+		manifests.WithImage(image),
+	)
+	if err != nil {
+		return err
 	}
-	if !ensureCSV(log, image, csv, highAvailabilityMode) {
+
+	if !ensureCSV(log, image, expectedCSV, highAvailabilityMode) {
 		log.V(3).Info("no further updates are necessary to the packageserver csv")
 	}
 
+	for k, v := range expectedCSV.GetLabels() {
+		csv.ObjectMeta.Labels[k] = v
+	}
+
+	for k, v := range expectedCSV.GetAnnotations() {
+		csv.ObjectMeta.Annotations[k] = v
+	}
+
+	csv.Spec = expectedCSV.Spec
```

Comment:
So basically, unless our narrowly-defined [validateCsv](https://github.com/openshift/operator-framework-olm/blob/fceaf8aba28fb5c9c8f636c49b0fe3119d7d348d/pkg/package-server-manager/config.go#L107-L121) failed, we would not consider what the default contained.  

And then you're adding stickiness to the existing labels and annotations, which should address the original complaint.  

The only question I have is that in the old code we accumulated to `csv.Spec` before invoking `ensureCSV` and now we do it afterwards.  Was the reordering intentional?

Reply:
That seems reasonable to me

Reply:
I think we'd want to add the annotations and labels to ensureCSV as well.

Reply:
To be clear, I generated the expectedCSV first so the ensureCSV function could set the expected values, but it seems reasonable to move the creation of the expectedCSV within the ensureCSV method.

Reply:
And the expectedCSV is passed into the ensureCSV method before the cab.Spec is set

Reply:
Hey! I've added the tests you requested!